spec/features/course/unread_status_management_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe 'Announcements read/unread status management', type: :feature do
+  subject { page }
+
+  let!(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+
+    let(:user) { create(:user, role: :administrator) }
+    let(:course) { create(:course) }
+
+    before do
+      login_as(user, scope: :user)
+    end
+
+    context 'with a new announcement' do
+      let!(:announcement) { create(:course_announcement, course: course) }
+
+      it 'increases unread announcement number by 1' do
+        visit course_achievements_path(course)
+        expect(page).to have_link('Announcements (1)')
+      end
+    end
+
+    context 'with five new announcements' do
+      let!(:announcements) { create_list(:course_announcement, 5, course: course) }
+
+      it 'increases unread announcement number by 5' do
+        visit course_achievements_path(course)
+        expect(page).to have_link('Announcements (5)')
+      end
+    end
+
+    context 'when visit announcements index page' do
+      let!(:announcements) { create_list(:course_announcement, 5, course: course) }
+
+      it 'marks unread announcement number as 0' do
+        visit course_announcements_path(course)
+        expect(page).to have_link('Announcements (0)')
+      end
+    end
+
+    context 'when create an announcement for one course and visit another course page' do
+      let!(:announcement) { create(:course_announcement, course: course) }
+      let!(:course_2) { create(:course) }
+
+      it 'does not change unread announcement number of another course' do
+        visit course_achievements_path(course_2)
+        expect(page).to have_link('Announcements (0)')
+      end
+    end
+
+    context 'when user create an announcement for a course himself' do
+      let(:announcement) { build(:course_announcement, course: course) }
+
+      before do
+        visit new_course_announcement_path(course)
+        fill_in 'course_announcement_title',    with: announcement.title
+        fill_in 'course_announcement_content',  with: announcement.content
+      end
+
+      it 'does not change his unread announcement number of this course' do
+        expect { click_button 'Create' }.to change(Course::Announcement, :count).by(1)
```

Comment:
Line is too long. [86/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.